tests/test_routes.py

```diff
@@ -530,4 +537,246 @@ def test_create_item_duplicate_id(self):
         )
         self.assertEqual(resp.status_code, status.HTTP_409_CONFLICT)
     
-    
\ No newline at end of file
+
+    ######################################################################
+    # TEST UPDATE ITEM
+    ######################################################################
+    def test_update_item_no_quantity_no_price(self):
+        """ Attempts updating an item with no quantity and no price in body. """
+        # create an item first
+        test_item = ItemFactory()
+        req = test_item.serialize()
+        user_id = req.pop("user_id")
```

Comment:
For all of the tests I believe you should pop item_name too. The factory creates an item with 5 attributes: user_id, item_id, item_name, quantity, price. Need to pop 3 or 4

Reply:
Nah. Item name does not matter at all. In the request body, you can include the item_name or discard it. In my code logic, no matter you included the item_name or not, it does not matter. Only quantity and price will take effect.
See the screenshot below:
<img width="643" alt="Screen Shot 2022-04-03 at 5 47 12 PM" src="https://user-images.githubusercontent.com/37674309/161450109-b8470c85-8193-4171-bee5-226344b8bee6.png">

Reply:
I agree that it doesn't make a difference because we know the implementation haha. But I meant if we want to really test our feature against the acceptance criteria, we should only pass in quantity or price haha, as a real user world, for unit testing.

Reply:
Actually I am not sure about this part.. It is a good question that helps us understand the unit testing. Thanks for forwarding it. Let me check with professor on Slack.

Reply:
In the ZenHub story for update_items, we defined the acceptance criteria to say the inputs should only be {price, quantity}. Of course we can alter this be {item_name, price, quantity} but then we would need to be clear about this in the entire code (either that item_name is updated too or item_name is simply discarded)

Reply:
My understanding is that we should make item_name optional, users can have it, can not have it, it depends on the user

Reply:
I would suggest changing the acceptance criteria.

Reply:
No problem. But just to be clear, in the current code implementation, whatever the item_name passed in is, it is simply discarded. Is this the behaviour we want? So we can pass in a different value for item_name and we won't have any idea that it wasn't changed but simply discarded.

Reply:
Actually I also think it depends on the front end -- if we want to include 'item_name' field in UI when we are performing the update item

Reply:
I haven't check what UI we will have yet, anyways, I will remove 'item_name' in unit testing for now

Reply:
I think we can make the UI whatever we want, either allow us to change item_name or not. But regardless, since item_id is a unique identifier already, we don't need item_name to confirm the resource; so I think we should either not use it and say clearly that it won't be used even if passed, or allow changing names too